vllm/config.py

```diff
@@ -955,9 +955,16 @@ def __init__(self,
                  send_delta_data: bool = False) -> None:
         if max_num_batched_tokens is None:
             if enable_chunked_prefill:
-                # It is the values that have the best balance between ITL
-                # and TTFT on A100. Note it is not optimized for throughput.
-                max_num_batched_tokens = 512
+                if num_scheduler_steps > 1:
+                    # Multi-step Chunked-Prefill doesn't allow prompt-chunking
+                    # for now. Have max_num_batched_tokens set to max_model_len
+                    # so we don't reject sequences on account of a short
+                    # max_num_batched_tokens.
+                    max_num_batched_tokens = max(max_model_len, 2048)
```

Comment:
I'll add a follow-up PR to remove this condition. The implementation of Chunking prompts isn't hard (I have an implementation that supports prompt chunking [here](https://github.com/vllm-project/vllm/pull/8001/files#diff-2df570790076c0f655d43fb01b48615f7eb4c99dc0bfeef157d898c468770e53). The implementation chunks the prompt and processes one chunk in one multi-step.  

 There are some considerations though: 
  1. Prompt chunking has some performance implications:
     - More `execute_model` steps will in eager mode as the steps wouldn't be pure decode.
     - Some benchmarking on the closed PR mentioned above, suggested no performance gain. It might be  due to the how it was implemented. I'll have to look into it.
  2. Prompt chunking with multi-step involves more state tracking.
      -  It breaks the current assumption that all queries are decodes in advance_step.
      - Need to update `seq_start_loc`, `context_lens` tensors in advance_step for prefill attention
      - Need to update the input_tokens with the next set of prefill input_tokens
 
  Given the performance and code-change implications, it only makes sense to do it if the performance gains are reasonable. I'll work on this as a follow up.

Reply:
This only happens when the user hasn't specified a `max-num-batched-tokens` value. A logging exists already at https://github.com/vllm-project/vllm/blob/9e5ec35b1f8239453b1aaab28e7a02307db4ab1f/vllm/config.py#L984